api/notd/token_listing_processor.py

```diff
@@ -97,9 +97,29 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
                     sortedAssetListings = sorted(assetListings, key=lambda listing: listing.value, reverse=False)
                     listings.append(sortedAssetListings[0])
             # NOTE(krishan711): sleep to avoid opensea limits
-            await asyncio.sleep(0.1)
+            await asyncio.sleep(0.25)
         return listings
 
+    async def get_changed_opensea_token_listings_for_collection(self, address: str, startDate: datetime.datetime) -> List[str]:
+        queryData = {
+            'asset_contract_address': address,
+            'occurred_after': startDate,
+        }
+        tokensIdsToReprocess = set()
+        while True:
+            response = await self.openseaRequester.get(url="https://api.opensea.io/api/v1/events", dataDict=queryData, timeout=30)
+            responseJson = response.json()
+            logging.info(f'Got {len(responseJson["asset_events"])} items')
+            for asset in responseJson['asset_events']:
+                if asset['asset'] and asset.get('event_type') != 'bid_entered':
```

Comment:
```suggestion
                if asset['asset'] and asset.get('event_type') not in {'bid_entered', ...}:
```

Reply:
also why do you have `asset['asset'] ` here? in what case would it not be there?

Reply:
when I started testing there were some responses it wasn't present.

Reply:
{'asset': None, 'asset_bundle': None, 'event_type': 'collection_offer', 'event_timestamp': '2022-08-03T14:49:56.086934', 'auction_type': None, 'total_price': None, 'payment_token': {'symbol': 'WETH', 'address': '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', 'image_url': 'https://openseauserdata.com/files/accae6b6fb3888cbff27a013729c22dc.svg', 'name': 'Wrapped Ether', 'decimals': 18, 'eth_price': '1.000000000000000', 'usd_price': '1406.910000000000082000'}, 'transaction': None, 'created_date': '2022-08-03T14:49:56.088046', 'quantity': '1', 'approved_account': None, 'bid_amount': '1691500000000000000', 'collection_slug': 'goblintownwtf', 'contract_address': '', 'custom_event_name': None, 'dev_fee_payment_event': None, 'dev_seller_fee_basis_points': 750, 'duration': None, 'ending_price': '1691500000000000000', 'from_account': {'user': None, 'profile_img_url': 'https://storage.googleapis.com/opensea-static/opensea-profile/31.png', 'address': '0xf4ec5185cde9ec7b1e0bbd0095d0268aa77a1ae6', 'config': ''}, 'id': 7279020849, 'is_private': None, 'owner_account': None, 'seller': None, 'starting_price': '1691500000000000000', 'to_account': None, 'winner_account': None, 'listing_time': None}

Reply:
cool thanks, we should ignore `collection_offer` as well